src/consensus/vote_parser.zig

```diff
@@ -0,0 +1,126 @@
+//! Based on https://github.com/anza-xyz/agave/blob/182823ee353ee64fde230dbad96d8e24b6cd065a/vote/src/vote_parser.rs
+
+// TODO: this is probably/definitely the wrong place for this file to be,
+// but it's the only place it's needed right now, and I don't feel like
+// committing to a solid structure yet.
+
+const std = @import("std");
+const sig = @import("../sig.zig");
+
+const vote_program = sig.runtime.program.vote_program;
+
+const Hash = sig.core.Hash;
+const Pubkey = sig.core.Pubkey;
+const Signature = sig.core.Signature;
+const Transaction = sig.core.Transaction;
+
+const VoteTransaction = sig.consensus.vote_transaction.VoteTransaction;
+
+pub const ParsedVote = struct {
+    key: Pubkey,
+    vote: VoteTransaction,
+    switch_proof_hash: ?Hash,
+    signature: Signature,
+
+    pub fn deinit(self: ParsedVote, allocator: std.mem.Allocator) void {
+        self.vote.deinit(allocator);
+    }
+};
+
+// Used for parsing gossip vote transactions
+pub fn parseVoteTransaction(
+    allocator: std.mem.Allocator,
+    tx: Transaction,
+) std.mem.Allocator.Error!?ParsedVote {
+    // Check first instruction for a vote
+    const message = tx.msg;
+
+    if (message.instructions.len == 0) return null;
+    const first_instruction = message.instructions[0];
```

Comment:
Is this a requirement for vote transactions? It must be in the first instruction?

Reply:
Very much appears so: https://github.com/anza-xyz/agave/blob/640a749ea599a05a1e2e553ed70f835752cb68e8/vote/src/vote_parser.rs#L25-L39